src/python/pants/backend/python/subsystems/pex_build_util.py

```diff
@@ -87,7 +92,30 @@ def dump_sources(builder, tgt, log):
 class PexBuilderWrapper(object):
   """Wraps PEXBuilder to provide an API that consumes targets and other BUILD file entities."""
 
-  def __init__(self, builder, python_repos_subsystem, python_setup_subsystem, log=None):
+  class Factory(Subsystem):
```

Comment:
Can we revisit whether the `Factory` is necessary in these cases? Is the idea behind the `Factory` that the `Factory` receives the relevant scoped `Options`, but that the thing the subsystem constructs might have some additional steps involved in initialization?

I've noticed that in some of those cases, using `@memoized_(property|method)` to memoize the additional construction steps can allow for avoiding the split between an explicit Factory/Builder and the built thing.

Reply:
Subsystem has constructor:
https://github.com/pantsbuild/pants/blob/039051735542d29ae02f4faa09c0c51c47292bf0/src/python/pants/subsystem/subsystem.py#L134

I see no non-hacky way to make a class that needs more than options in its constructor be a Subsystem. Worse in this case, the additional constructor options are not global in character or even scoped in character - they are per-use. IOW there are existing use cases in this review where a task creates two different pexs using the PexBuilderWrapper - each needs its own PEXBuilder if not log. So the factory truly is a single factory built from its options (empty here - a wart of the combination of options with deps) and deps that then produces a unique PexBuilderWrapper per PEXBuilder.

Reply:
> Worse in this case, the additional constructor options are not global in character or even scoped in character - they are per-use.

Ok. So in this case the factory is less about the fact that a `Subsystem` is involed, and more about the fact that what is being configured is not an instance of a tool, but literally a Factory/Builder that needs more information. Makes sense.

> I see no non-hacky way to make a class that needs more than options in its constructor be a Subsystem.

So then, in some other hypothetical case where all the Subsystem needed was options in order to be constructed, the Factory might not be necessary? If so, that makes sense... I don't think I was making the distinction between "needs only options to be constructed" and "needs options and some other stuff" to be constructed.

Reply:
Exactly. A Subsystem, by design, can only require options and other subsystem dependencies to be constructed.

Reply:
This was a useful thread for me to see as I was not super clear on when to model new code after the existing `.Factory` classes throughout the codebase. You can see in the native backend that I have been making quite a number of subsystems and composing them, which may have led to some more complex logic than necessary.